org.adempiere.base/src/org/compiere/model/MInvoice.java

```diff
@@ -2319,8 +2321,10 @@ else if (getGrandTotal().compareTo(totalPayAmt) != 0 && ptList.size() > 0)
 				}
 			}
 		}
-
-		if (PAYMENTRULE_Cash.equals(getPaymentRule())) {
+		MOrder order = new MOrder(getCtx(), getC_Order_ID(), get_TrxName());
+		MDocType orderDocType = new MDocType(getCtx(), order.getC_DocType_ID(), get_TrxName());

```

Comment:
for better performance, should use `MDocType.get(order.getC_DocType_ID())` instead